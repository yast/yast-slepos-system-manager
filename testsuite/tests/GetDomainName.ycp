{

// testedfiles: SleposSystemManger.ycp

include "testsuite.ycp";
TESTSUITE_INIT([], nil);

import "SleposSystemManager";

integer testNum = 0;

map READ = $[];
map WRITE = $[];
map EXECUTE_0 = $[
	"target" : $[
		"bash_output" : $["exit":0, "stdout":"stdout messages", "stderr":"An error message!"]
	]
];

map EXECUTE_1 = $[
	"target" : $[
		"bash_output" : $["exit":1, "stdout":"stdout messages", "stderr":"An error message!"]
	]
];


// Test for nil param
testNum = testNum + 1;
DUMP("*******************************************************");
DUMP(sformat("TEST CASE #%1", testNum));
string DN = nil;
DUMP(sformat("SleposSystemManager::GetDomainName(%1)", DN));
TEST(``(SleposSystemManager::GetDomainName(DN)), [READ,WRITE,EXECUTE_0], nil);

// Test for empty string param
testNum = testNum + 1;
DUMP("*******************************************************");
DUMP(sformat("TEST CASE #%1", testNum));
DN = "";
DUMP(sformat("SleposSystemManager::GetDomainName(%1)", DN));
TEST(``(SleposSystemManager::GetDomainName(DN)), [READ,WRITE,EXECUTE_0], nil);

// Test c=us
testNum = testNum + 1;
DUMP("*******************************************************");
DUMP(sformat("TEST CASE #%1", testNum));
DN = "c=us";
DUMP(sformat("SleposSystemManager::GetDomainName(%1)", DN));
TEST(``(SleposSystemManager::GetDomainName(DN)), [READ,WRITE,EXECUTE_0], nil);

// Test o=novell,c=us
testNum = testNum + 1;
DUMP("*******************************************************");
DUMP(sformat("TEST CASE #%1", testNum));
DN = "o=novell,c=us";
DUMP(sformat("SleposSystemManager::GetDomainName(%1)", DN));
TEST(``(SleposSystemManager::GetDomainName(DN)), [READ,WRITE,EXECUTE_0], nil);

// Test ou=ut,o=novell,c=us
testNum = testNum + 1;
DUMP("*******************************************************");
DUMP(sformat("TEST CASE #%1", testNum));
DN = "ou=ut,o=novell,c=us";
DUMP(sformat("SleposSystemManager::GetDomainName(%1)", DN));
TEST(``(SleposSystemManager::GetDomainName(DN)), [READ,WRITE,EXECUTE_0], nil);

// Test ou=provo,ou=ut,o=novell,c=us
testNum = testNum + 1;
DUMP("*******************************************************");
DUMP(sformat("TEST CASE #%1", testNum));
DN = "ou=provo,ou=ut,o=novell,c=us";
DUMP(sformat("SleposSystemManager::GetDomainName(%1)", DN));
TEST(``(SleposSystemManager::GetDomainName(DN)), [READ,WRITE,EXECUTE_0], nil);

// Test cn=bld_H,ou=provo,ou=provo,ou=ut,o=novell,c=us
testNum = testNum + 1;
DUMP("*******************************************************");
DUMP(sformat("TEST CASE #%1", testNum));
DN = "cn=bld_H,ou=provo,ou=ut,o=novell,c=us";
DUMP(sformat("SleposSystemManager::GetDomainName(%1)", DN));
TEST(``(SleposSystemManager::GetDomainName(DN)), [READ,WRITE,EXECUTE_0], nil);

// Test cn=bs-21,cn=server,cn=bld_H,ou=provo,ou=provo,ou=ut,o=novell,c=us
testNum = testNum + 1;
DUMP("*******************************************************");
DUMP(sformat("TEST CASE #%1", testNum));
DN = "cn=bs-21,cn=server,cn=bld_H,ou=provo,ou=ut,o=novell,c=us";
DUMP(sformat("SleposSystemManager::GetDomainName(%1)", DN));
TEST(``(SleposSystemManager::GetDomainName(DN)), [READ,WRITE,EXECUTE_0], nil);

// Test cn=50.1,cn=bs-21,cn=server,cn=bld_H,ou=provo,ou=provo,ou=ut,o=novell,c=us
testNum = testNum + 1;
DUMP("*******************************************************");
DUMP(sformat("TEST CASE #%1", testNum));
DN = "cn=50.1,cn=bs-21,cn=server,cn=bld_H,ou=provo,ou=ut,o=novell,c=us";
DUMP(sformat("SleposSystemManager::GetDomainName(%1)", DN));
TEST(``(SleposSystemManager::GetDomainName(DN)), [READ,WRITE,EXECUTE_0], nil);

// Test cn=50.1
testNum = testNum + 1;
DUMP("*******************************************************");
DUMP(sformat("TEST CASE #%1", testNum));
DN = "cn=50.1";
DUMP(sformat("SleposSystemManager::GetDomainName(%1)", DN));
TEST(``(SleposSystemManager::GetDomainName(DN)), [READ,WRITE,EXECUTE_0], nil);

}
