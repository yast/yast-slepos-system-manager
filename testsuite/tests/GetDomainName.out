Dump	*******************************************************
Dump	TEST CASE #1
Dump	SleposSystemManager::GetDomainName(nil)
Return	
Dump	*******************************************************
Dump	TEST CASE #2
Dump	SleposSystemManager::GetDomainName()
Return	
Dump	*******************************************************
Dump	TEST CASE #3
Dump	SleposSystemManager::GetDomainName(c=us)
Return	us
Dump	*******************************************************
Dump	TEST CASE #4
Dump	SleposSystemManager::GetDomainName(o=novell,c=us)
Return	novell.us
Dump	*******************************************************
Dump	TEST CASE #5
Dump	SleposSystemManager::GetDomainName(ou=ut,o=novell,c=us)
Return	ut.novell.us
Dump	*******************************************************
Dump	TEST CASE #6
Dump	SleposSystemManager::GetDomainName(ou=provo,ou=ut,o=novell,c=us)
Return	provo.ut.novell.us
Dump	*******************************************************
Dump	TEST CASE #7
Dump	SleposSystemManager::GetDomainName(cn=bld_H,ou=provo,ou=ut,o=novell,c=us)
Return	bld_H.provo.ut.novell.us
Dump	*******************************************************
Dump	TEST CASE #8
Dump	SleposSystemManager::GetDomainName(cn=bs-21,cn=server,cn=bld_H,ou=provo,ou=ut,o=novell,c=us)
Return	bs-21.server.bld_H.provo.ut.novell.us
Dump	*******************************************************
Dump	TEST CASE #9
Dump	SleposSystemManager::GetDomainName(cn=50.1,cn=bs-21,cn=server,cn=bld_H,ou=provo,ou=ut,o=novell,c=us)
Return	50.1.bs-21.server.bld_H.provo.ut.novell.us
Dump	*******************************************************
Dump	TEST CASE #10
Dump	SleposSystemManager::GetDomainName(cn=50.1)
Return	50.1
